apps/meteor/client/views/admin/moderation/UserMessages.tsx

```diff
@@ -0,0 +1,126 @@
+import { Box, Callout, Message, States, StatesIcon, StatesTitle } from '@rocket.chat/fuselage';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useEndpoint, useRoute, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import React, { useMemo } from 'react';
+
+import VerticalBar from '../../../components/VerticalBar';
+import { useUserDisplayName } from '../../../hooks/useUserDisplayName';
+import MessageContextFooter from './MessageContextFooter';
+import ContextMessage from './helpers/ContextMessage';
+
+const UserMessages = ({
+	userId,
+	reload,
+	onRedirect,
+}: {
+	userId: string;
+	reload: () => void;
+	onRedirect: (mid: string) => void;
+}): JSX.Element => {
+	const t = useTranslation();
+
+	const moderationRoute = useRoute('moderation-console');
+
+	const getUserMessages = useEndpoint('GET', '/v1/moderation.user.reportedMessages');
+
+	const query = useMemo(() => ({ userId }), [userId]);
```

Comment:
there is no need to memoize useId here, useQuery evaluate prop by prop so its not going to invalidate creating a new object